### PR TITLE
feat(term): add term/conpty

### PR DESCRIPTION
**Pull request by aymanbagabas**
Add support for windows conpty https://learn.microsoft.com/en-us/windows/console/creating-a-pseudoconsole-session

### CODE REVIEW

File: term/conpty/conpty_windows.go [hunk 1]
```diff
@@ -0,0 +1,271 @@
+package conpty
```
Comment (caarlos0) [inline term/conpty/conpty_windows.go:1]: Should we add `go:build windows`?

**Review by caarlos0 — COMMENTED** [review]
Lgtm, probably good to re-run the scripts that gen the CI scripts too

File: term/go.mod [hunk 2]
```diff
@@ -0,0 +1,5 @@
+module github.com/charmbracelet/x/term
+
+go 1.21
```
Comment (caarlos0) [inline term/go.mod:3]: can we lower this?

Comment (aymanbagabas) [reply term/go.mod]: done

**Review by caarlos0 — APPROVED** [review]
NOICE

**Pull request merged**